sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalSuite.scala

```diff
@@ -122,4 +126,25 @@ class SimplifyConditionalSuite extends PlanTest with PredicateHelper {
         None),
       CaseWhen(normalBranch :: trueBranch :: Nil, None))
   }
+
+  test("remove entire CaseWhen if all the outputs are semantic equivalence") {
```

Comment:
We may need test case including non deterministic cond.

Reply:
Yes, I plan to add couple more tests tonight.